torch/csrc/jit/passes/onnx/constant_fold.cpp

```diff
@@ -0,0 +1,187 @@
+#include <torch/csrc/jit/passes/onnx/constant_fold.h>
+#include <c10/util/Exception.h>
+
+#include <c10/util/Optional.h>
+#include <algorithm> 
+
+#if defined(_MSC_VER)
+#include <BaseTsd.h>
+typedef SSIZE_T ssize_t;
+#endif
+
+namespace torch {
+namespace jit {
+
+namespace onnx {
+using namespace ::c10::onnx;
+}
+
+using ParamMap = std::map<std::string, at::Tensor>;
+using ValueToParamPairMap = std::map<Value*, std::pair<std::string, at::Tensor>>;
+
+static ValueToParamPairMap buildValueToParamsMap(Block* b, const ParamMap& paramsDict) {
+  ValueToParamPairMap valsToParamsMap;
+  for(auto& input : b->inputs()) {
+      auto it = paramsDict.find(input->uniqueName());
+      if (it != paramsDict.end()) {
+          valsToParamsMap[input] = *it;
+      }
+  }
+  return valsToParamsMap;
+}
+
+static void buildParamsMapFromValueToParamsMap(
+    const ValueToParamPairMap& valsToParamsMap, ParamMap& paramsDict) {
+  paramsDict.clear();
+  for(auto& nameTensorParamPair : valsToParamsMap) {
+    paramsDict.insert(nameTensorParamPair.second);
+  }
+}
+
+static void eraseUnusedNodeOutputs(Node* node) {
+  std::vector<std::string> removedOutputNames;
+  for (size_t i_1 = node->outputs().size(); i_1 > 0; --i_1) {
+      size_t i = i_1 - 1;
+      if (!node->outputs().at(i)->hasUses()) {
+        node->eraseOutput(i);
+      }
+  }
+}
+
+static at::Tensor runTorchBackendForOnnx(const Node* node, std::vector<at::Tensor>& inputTensorValues) {
+  at::Tensor updated_val;
```

Comment:
@houseroad, can you take a look at these cases? I'm not familiar with the onnx->aten semantics

Reply:
Yes, that will be great. Just fyi - I have tested this part on multiple large production models, including BERT, to make sure that these computations are correct, and the numbers match exactly.

Reply:
Updated based on @houseroad 's feedback on this section.